bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/FlatRowAdapter.java

```diff
@@ -92,16 +89,31 @@ public FlatRow adaptToRow(Result result) {
     final Cell[] rawCells = result.rawCells();
     if (rawCells != null && rawCells.length > 0) {
       for (Cell rawCell : rawCells) {
-        rowBuilder.addCell(
-            Bytes.toString(
-                rawCell.getFamilyArray(), rawCell.getFamilyOffset(), rawCell.getFamilyLength()),
-            ByteStringer.wrap(
-                rawCell.getQualifierArray(),
-                rawCell.getQualifierOffset(),
-                rawCell.getQualifierLength()),
-            TimestampConverter.hbase2bigtable(rawCell.getTimestamp()),
-            ByteStringer.wrap(
-                rawCell.getValueArray(), rawCell.getValueOffset(), rawCell.getValueLength()));
+        if (rawCell instanceof RowCell) {
```

Comment:
can you extract the commonalities? It's really hard to see the difference for the cases:

```
Cell.Builder cellBuilder = Cell.newBuilder()
  .withFamily(
    Bytes.toString(
      rowCell.getFamilyArray(),
      rowCell.getFamilyOffset(), rowCell.getFamilyLength())
  .withQualifier(....)
  ...;

if (rawCell instanceof RowCell)  {
   cellBuilder.withLabel(((RowCell)rowCell).getLabels())
}
```

Also we really should move this method to a test